pkg/services/publicdashboards/api/api.go

```diff
@@ -63,9 +63,11 @@ func (api *Api) RegisterAPIEndpoints() {
 	api.RouteRegister.Get("/api/public/dashboards/:accessToken", routing.Wrap(api.GetPublicDashboard))
 	api.RouteRegister.Post("/api/public/dashboards/:accessToken/panels/:panelId/query", routing.Wrap(api.QueryPublicDashboard))
 
+	// List Public Dashboards
+	api.RouteRegister.Get("/api/dashboards/public", auth(middleware.ReqSignedIn, accesscontrol.EvalPermission(dashboards.ActionDashboardsWrite)), routing.Wrap(api.ListPublicDashboards))
```

Comment:
Why do they need dashboard write permissions? Would have thought it would be dashboard read.

Reply:
This is a good point. In the beginning we thought this would be an admin option but if we are adding this into the dashboard menu we should retrieve all public dashboards associated with a dashboard that a user has read access to, using the scope. Same permissions as `GET /api/dashboards/uid/:uid/public-config` but we need to evaluate the scope for each row returned from public-dashboards.
In the future, if we put actions here (e.g. disabling) we should hide the button if the user doesn't have write permissions for a public dashboard.

Reply:
This was a typo from me- good catch. We decided that since users can see all public dashboards, they should be able to view all public dashboards on the instance of grafana for the current org.

Reply:
All users can see the public dashboards but not all users can see the public dashboards config and the original dashboard. If we decide to show all public dashboards here we need to add some logic to hide the buttons to the original dashboard and to the public config (links could be broken if the user doesn't have access to the dashboard)